neptune/internal/abort.py

```diff
@@ -34,56 +37,36 @@ def __init__(self, pid):
 
     @staticmethod
     def requirements_installed():
-        # pylint:disable=unused-import,unused-variable,bad-option-value,import-outside-toplevel
-        try:
-            import psutil
-            return True
-        except ImportError:
-            _logger.warning('psutil is not installed. The experiment will not be abortable.')
-            return False
+        return PSUTIL_INSTALLED
 
     def abort(self):
-        # pylint:disable=bad-option-value,import-outside-toplevel
-        import psutil
-
-        process = None
         try:
-            process = psutil.Process(self._pid)
+            processes = self._get_process_with_children(psutil.Process(self._pid))
         except psutil.NoSuchProcess:
-            pass
+            processes = []
 
-        if process is not None:
-            processes = self._get_processes(process)
-            for p in processes:
-                self._abort(p)
-            _, alive = psutil.wait_procs(processes, timeout=self.KILL_TIMEOUT)
-            for p in alive:
-                self._kill(p)
+        for p in processes:
+            self._abort(p)
+        _, alive = psutil.wait_procs(processes, timeout=self.KILL_TIMEOUT)
+        for p in alive:
+            self._kill(p)
 
     @staticmethod
-    def _get_processes(process):
-        # pylint:disable=bad-option-value,import-outside-toplevel
-        import psutil
-
+    def _get_process_with_children(process):
         try:
             return [process] + process.children(recursive=True)
         except psutil.NoSuchProcess:
             return []
 
-    def _abort(self, process):
-        # pylint:disable=bad-option-value,import-outside-toplevel
-        import psutil
-
+    @staticmethod
+    def _abort(process):
         try:
             process.terminate()
         except psutil.NoSuchProcess:
             pass
 
     def _kill(self, process):
-        # pylint:disable=bad-option-value,import-outside-toplevel
-        import psutil
-
-        for process in self._get_processes(process):
+        for process in self._get_process_with_children(process):
```

Comment:
I think the author's idea might have been that since then some new children could've popped up. This, of course, is far from bulletproof. Just a bit better.